content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -1,48 +1,75 @@
-This is an article about Sockets. It will give you a glimpse of how clients and servers communicate over the internet.
+This article is about Sockets. It will give you a glimpse of how clients and servers communicate over the internet.
+
 ### Introduction
-Socket programming is a way of connecting two computers over a network to allow the exchange of information.The connection can be through **connection-oriented protocol** or **connectionless**. In our case, we will be using  TCP/IP a connection-oriented protocol.
-In connection-oriented protocols, two computers must establish a connection before transferring data. Whereas in connectionless, you do not need every packet to arrive at its desired destination. This is made possible by UDP(User Datagram Protocol)
-We will be using the Client\Server architecture to illustrate more on sockets. Client and server communicate by writing to and reading from the socket connection. To establish this connection, the machines must have each other's network information.
+
+Socket programming is a means of communicating data between two computers across a network. Connections can be made using either a **connection-oriented protocol** or a **connectionless protocol**. TCP/IP, a connection-oriented protocol, will be used in our case.
+
+Before exchanging data, two computers must establish a link in connection-oriented protocols. In contrast, you don't require every packet to arrive at its intended destination for connectionless. It is made feasible via UDP (User Datagram Protocol). To demonstrate sockets further, we shall use the Client\Server architecture. Client and server communicate by writing to and reading from the socket connection. The machines must have each other's network information to make this connection.
+
 Table of contents:
 - [Definition-of-a-Socket](#definition-of-a-socket)
```

Comment:
You didn't implement the suggestion I made last time here (the suggestion I made should still be visible here). Please implement that suggestion or let me know why you don't want to do so.

Reply:
In case you can't see it I wanted you to remove the *-*'s between the words within the square brackets (these: **[]**) so change:
> - \[Definition-of-a-Socket\]\(#definition-of-a-socket)
> - \[Server-Side-Programming\ ](#server-side-programming)
> - \[Client-Side-Programming\](#client-side-programming)
> - \[How-to-Run-the-Programs-on-different-applications\](#how-to-run-the-programs-on-diffrent-applications)
> - \[Conclusion\](#conclusion)

To:

> - \[Definition of a Socket\](#definition-of-a-socket)
> - \[Server Side Programming\](#server-side-programming)
> - \[Client Side Programming\](#client-side-programming)
> - \[How to Run the Programs on different applications\](#how-to-run-the-programs-on-diffrent-applications)
> - \[Conclusion\](#conclusion)